CIS 600 Data Mining
Project
--------
Import Libraries

In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.cm as cm
import seaborn as sns


Import source data

In [25]:
URL = 'https://raw.githubusercontent.com/SteffiPeTaffy/machineLearningAZ/refs/heads/master/Machine%20Learning%20A-Z%20Template%20Folder/Part%204%20-%20Clustering/Section%2025%20-%20Hierarchical%20Clustering/Mall_Customers.csv'
df = pd.read_csv(URL)

print('Dataset loaded successfully!')
print(f'Shape: {df.shape[0]} rows × {df.shape[1]} columns')
df.head(10)

Dataset loaded successfully!
Shape: 200 rows × 5 columns


,CustomerID,Genre,Age,Annual Income (k$),Spending Score (1-100)
0,1,Male,19,15,39
1,2,Male,21,15,81
2,3,Female,20,16,6
3,4,Female,23,16,77
4,5,Female,31,17,40
5,6,Female,22,17,76
6,7,Female,35,18,6
7,8,Female,23,18,94
8,9,Male,64,19,3
9,10,Female,30,19,72


Step 1
Data Understanding (10 points)


In [26]:
df.info()
df.describe()

<class 'pandas.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype
---  ------                  --------------  -----
 0   CustomerID              200 non-null    int64
 1   Genre                   200 non-null    str  
 2   Age                     200 non-null    int64
 3   Annual Income (k$)      200 non-null    int64
 4   Spending Score (1-100)  200 non-null    int64
dtypes: int64(4), str(1)
memory usage: 7.9 KB


,CustomerID,Age,Annual Income (k$),Spending Score (1-100)
count,200.000000,200.000000,200.000000,200.000000
mean,100.500000,38.850000,60.560000,50.200000
std,57.879185,13.969007,26.264721,25.823522
min,1.000000,18.000000,15.000000,1.000000
25%,50.750000,28.750000,41.500000,34.750000
50%,100.500000,36.000000,61.500000,50.000000
75%,150.250000,49.000000,78.000000,73.000000
max,200.000000,70.000000,137.000000,99.000000


What does this data represent?: This data represents mall customer demographics gathered via membership card details.
Column 0: CustomerID, unique identifier, can be discarded for analysis purposes
Column 1: Genre, not sure why its labelled this way, but basically gender
Column 2: Age (Range 18 - 70)
Column 3: Annual Income in thousands (Range 15K to 137K)
Column 4: Spending Score, (1-100) (The higher the score the better the customer is. This is what we want to maximize) (Range 1 - 99)

There are 200 rows of data, all features have data, (200 Non-Null).

Since we are looking groups of customers with similar behaviors i.e. clusters, we need to encode the categorical feature (genre) to a numerical equivalent. Also drop the irrelevant column and finally rename the columns just because genre make no sense.



Step 2: Data Cleaning and Preparation (15 points)


In [27]:
# Rename the columns
df.columns = ['CustomerID', 'Gender', 'Age', 'Annual Income', 'Spending Score']
print('Renamed columns:', df.columns.tolist())

# Encode Gender Columns and then validate
df['Gender'] = df['Gender'].map({'Male': 1, 'Female': 0})
print('Gender value counts after encoding:')
print(df['Gender'].value_counts())

Renamed columns: ['CustomerID', 'Gender', 'Age', 'Annual Income', 'Spending Score']
Gender value counts after encoding:
Gender
0    112
1     88
Name: count, dtype: int64


First validate that the total is 200 (112 + 88 = 200), so we know there is no issue with this column's data.

Next drop the column that has no impact to what we are trying to find, i.e. CustomerID


In [28]:
# Check for any duplicate rows
print(f'Duplicate rows: {df.duplicated().sum()}')

Duplicate rows: 0


Next drop the column that has no impact on what we are trying to find, i.e. CustomerID

In [29]:
# Drop CustomerID Column
df_clean = df.drop(columns=['CustomerID'])